core/unix/signal_linux_x86.c

```diff
@@ -179,7 +179,8 @@ convert_fxsave_to_fpstate(kernel_fpstate_t *fpstate, struct i387_fxsave_struct *
     fpstate->status = fxsave->swd;
     fpstate->magic = X86_FXSR_MAGIC;
 
-    memcpy(&fpstate->_fxsr_env[0], fxsave, sizeof(struct i387_fxsave_struct));
+    memcpy(&fpstate->_fxsr_env[0], fxsave,
+           sizeof(struct i387_fxsave_struct) - sizeof(fpstate->sw_reserved));
```

Comment:
offsetof seems more robust: will handle appended fields.  Ditto below.

Reply:
I should be able to use xmm_space's offset.

Reply:
Ok. Changed to using offsetof(struct i387_fxsave_struct, xmm_space).
We can use the entire sizeof(i387_fsave_struct) for the fnsave case.

Reply:
It appears that newer kernel sources have clarified extended_size, and it seems to depict the extended xstate area's size as discussed offline:

```
* If sw_reserved.magic1 == FP_XSTATE_MAGIC1 then there's a
 * sw_reserved.extended_size bytes large extended context area present. (The
 * last 32-bit word of this extended area (at the
 * fpstate+extended_size-FP_XSTATE_MAGIC2_SIZE address) is set to
 * FP_XSTATE_MAGIC2 so that you can sanity check your size calculations.)
 *
 * This extended area typically grows with newer CPUs that have larger and
 * larger XSAVE areas.
```

I suggest updating our sigcontext.h file.

Reply:
Done.